src/hotspot/share/gc/shared/referenceProcessor.cpp

```diff
@@ -771,16 +793,31 @@ void ReferenceProcessor::balance_queues(DiscoveredList ref_lists[])
 #endif
 }
 
-void ReferenceProcessor::process_soft_ref_reconsider(BoolObjectClosure* is_alive,
-                                                     OopClosure* keep_alive,
-                                                     VoidClosure* complete_gc,
-                                                     AbstractRefProcTaskExecutor* task_executor,
-                                                     ReferenceProcessorPhaseTimes* phase_times) {
-  assert(!processing_is_mt() || task_executor != NULL, "Task executor must not be NULL when mt processing is set.");
+void ReferenceProcessor::run_task(AbstractGangTask& task, AbstractClosureContext& closure_context, bool marks_oops_alive) {
+  WorkGang* gang = Universe::heap()->safepoint_workers();
+  assert(gang != NULL || !processing_is_mt(), "can not dispatch multi threaded without a work gang");
+  log_debug(gc, ref)("ReferenceProcessor::execute queues: %d, %s, marks_oops_alive: %s",
+    num_queues(), processing_is_mt() ? "ThreadModel::Multi" : "ThreadModel::Single", marks_oops_alive?"true":"false");
```

Comment:
Indentation. One parameter per line please if they are long. Spaces between operators. Please also add a newline here so that the "boring" assertions are separated a bit more from the actual code.

Reply:
Will improve.